crates/ide-assists/src/handlers/extract_function.rs

```diff
@@ -676,11 +678,11 @@ impl FunctionBody {
                 parent_loop.get_or_insert(loop_.syntax().clone());
             }
         };
-        let (is_const, expr, ty) = loop {
+        let (is_const, expr, ty, generic_param_list, where_clause) = loop {
```

Comment:
In this case it might be better to create these as variables prior like `let mut generic_param_list = None;` and only set them in the appropriate branches as most do not set them actually.